### PR TITLE
optimize(t544): drop unsafe

### PR DESCRIPTION
**Pull request by fumiama**

### CODE REVIEW

**Comment by fumiama** [comment]
this may also fix #2068 #2045 #2044

File: internal/encryption/t544/encryption_amd64.s [hunk 1]
```diff
@@ -467,7 +468,7 @@ TEXT ·initState(SB), NOSPLIT, $0-64
     MOVUPS  X6,112(DI)
     RET
 
-TEXT ·sub_ad(SB), NOSPLIT, $8-24
+TEXT sub_ad(SB), NOSPLIT, $8-8
```
Comment (wdvxdr1123) [inline internal/encryption/t544/encryption_amd64.s:471]: TEXT sub_ad<>(SB), NOSPLIT, $8-8

FYI, `Adding <> to the name, as in foo<>(SB), makes the name visible only in the current source file, like a top-level static declaration in a C file. `

File: internal/encryption/t544/encryption_amd64.s [hunk 2]
```diff
@@ -610,6 +611,37 @@ TEXT ·sub_ad(SB), NOSPLIT, $8-24
...
 
+TEXT ·refreshState(SB), NOSPLIT, $16-8
+    MOVQ    ·i+0(FP), BX
+    MOVB    128(BX), CX
+    JE      ad
+    SHRQ    $1, CX
+fr:
+    MOVQ    BX, 0(SP)
+    MOVQ    CX, c-8(SP)
+    CALL    sub_ad(SB)
```
Comment (wdvxdr1123) [inline internal/encryption/t544/encryption_amd64.s:622]: CALL sub_ad<>(SB)

File: internal/encryption/t544/encryption_amd64.s [hunk 3]
```diff
@@ -610,6 +611,37 @@ TEXT ·sub_ad(SB), NOSPLIT, $8-24
     MOVUPS X0, 32(AX)
     RET
 
+TEXT ·refreshState(SB), NOSPLIT, $16-8
+    MOVQ    ·i+0(FP), BX
```
Comment (wdvxdr1123) [inline internal/encryption/t544/encryption_amd64.s:615]: s/·i/i

File: internal/encryption/t544/encryption_amd64.s [hunk 4]
```diff
@@ -236,20 +236,21 @@ TEXT ·sub_c(SB), NOSPLIT, $0-32
     MOVB CX, 15(AX)
     RET 
 
-TEXT ·sub_d(SB), NOSPLIT, $16-32
+TEXT ·sub_d(SB), NOSPLIT, $24-32
     MOVQ ·t+0(FP), BX
```
Comment (wdvxdr1123) [inline internal/encryption/t544/encryption_amd64.s:240]: s/·t/t

其他地方也一样

File: internal/encryption/t544/encryption.go [hunk 5]
```diff
@@ -50,4 +50,8 @@ func (c *state) init(key []byte, data []byte, counter uint64, nr uint8) {
 	initState(c, key, data, counter)
 }
 
-func sub_ad([]uint32)
+func refreshState(c *state)
+
+func (c *state) refreshState() {
```
Comment (wdvxdr1123) [inline internal/encryption/t544/encryption.go:55]: 看起来不需要这个方法，可以直接调用

**Comment by fumiama** [comment]
也许还需要测一下 #2070 有无复发（

File: internal/encryption/t544/encryption_amd64.s (hunk 1 above)
Comment (fumiama) [reply internal/encryption/t544/encryption_amd64.s:471]: that's good

File: internal/encryption/t544/encryption.go (hunk 5 above)
Comment (fumiama) [reply internal/encryption/t544/encryption.go:55]: sure

**Comment by fumiama** [comment]
all resolved

**Pull request merged**